common/scala/src/main/scala/org/apache/openwhisk/core/mesos/MesosTask.scala

```diff
@@ -142,9 +140,14 @@ object MesosTask {
           transid.finished(this, start, s"launched task ${taskId} at ${taskDetails.hostname}:${taskDetails
             .hostports(0)}", logLevel = InfoLevel)
         case Failure(ate: AskTimeoutException) =>
-          transid.failed(this, start, ate.getMessage, ErrorLevel)
+          transid.failed(this, start, s"task launch timed out ${ate.getMessage}", ErrorLevel)
           MetricEmitter.emitCounterMetric(LoggingMarkers.INVOKER_MESOS_CMD_TIMEOUT(LAUNCH_CMD))
-        case Failure(t) => transid.failed(this, start, t.getMessage, ErrorLevel)
+          //kill the task whose launch timed out
+          destroy(mesosClientActor, mesosConfig, taskId)
+        case Failure(t) =>
+          //kill the task whose launch timed out
+          destroy(mesosClientActor, mesosConfig, taskId)
```

Comment:
Note that the future returned by `destroy` for `andThen` call would be discarded so its more like ask and forget here. Is that the intention or it would be better to have destroy completed by the time `create`s resulting future completes?

Reply:
In this case destroy is just cleanup that is mesos-specific, so it should happen independent of the create future; i.e. `create()` timeout should return immediately, but trigger the cleanup of `destroy()`